src/utils/daos_hdlr.c

```diff
@@ -1645,6 +1646,80 @@ fs_copy_file(struct cmd_args_s *ap,
 	return rc;
 }
 
+#define SUB_STR_LEN 256
+static int
+fs_copy_symlink(struct cmd_args_s *ap,
+		struct file_dfs *src_file_dfs,
+		struct file_dfs *dst_file_dfs,
+		struct stat *src_stat,
+		const char *src_path,
+		const char *dst_path)
+{
+	int		rc;
+	daos_size_t	len = 0;
+	char		symlink_value[PATH_MAX-SUB_STR_LEN];
```

Comment:
Why PATH_MAX-SUB_STR_LEN here?  There doesn't seem to be a definition of the length of a symlink, so you should probably pick a initial value (perhaps 1k) and try that, then re-try if ENAMETOOLONG is returned.  If there's a limit in dfs then perhaps use that at the value and skip the retry, but if that's the case a comment here would be useful.

Reply:
The limit length of full path as well as symbolic link  value is PATH_NAX. Previously, symlink_value[PATH_MAX] leads to compiling error, "the frame size of bytes is larger than 4096". I just made changes to ignore warning -Wframe-larger-than= in src/utils/daos_hdlr.c. Now "char symlink_value[PATH_MAX]" is fine.

Reply:
you can just dynamically allocate the buffer

Reply:
True. I thought about dynamically allocating the buffer. Since the buffer size is small, I would prefer using local variable from stack to avoid the overhead from dynamic memory allocation.